target/hexagon/idef-parser/parser-helpers.c

```diff
@@ -427,6 +427,16 @@ int find_variable(Context *c, YYLTYPE *locp, HexValue *varid)
     return -1;
 }
 
+static unsigned find_variable_and_assert_declared(Context *c,
```

Comment:
You sure this functionality should not be embedded in `find_variable` entirely?
Seems like a reasonable safety check we always want.

Reply:
`find_variable` is only used in a single place (actually two but I'm removing one of them) and that is in the parser when declaring a new variable. It's needed to check that the variable isn't already declared, so it's good to have a version that doesn't assert. Unless we have some other way to check if a variable is already declared, I think it's best to keep both versions.

Reply:
I see, I think I'm just not super happy with the naming then.
If you have two versions of a function, one that can fail and one that cannot, usually you go for `try_get_something` and `get_something` (which calls `try_get_something` and `abort` in case of failure).

Reply:
Alright, I'll move it so `find_variable` asserts and `try_find_variable` can fail